solarforecastarbiter/reports/figures.py

```diff
@@ -327,23 +335,47 @@ def bar_subdivisions(cds, kind, metric):
     tools = 'pan,xwheel_zoom,box_zoom,box_select,reset,save'
     fig_kwargs = dict(tools=tools)
     figs = []
+    label_override = None
+
+    width = 0.8
+
+    # specific category/kind settings
     if kind == 'day':
+        fig_kwargs['x_axis_label'] = 'Day of the month'
+    elif kind == 'month':
+        fig_kwargs['x_axis_label'] = 'Month of the year'
+        label_override = [calendar.month_abbr[m] for m in cds.data[kind]]
+    elif kind == 'hour':
+        fig_kwargs['x_axis_label'] = 'Hour of the day'
+    elif kind == 'date':
+        fig_kwargs['x_axis_label'] = 'Date'
         fig_kwargs['x_axis_type'] = 'datetime'
-        width = 0.8 * pd.Timedelta('1day')
-    else:
-        width = 0.8
-
+        width = width * pd.Timedelta(days=1)
+    elif kind == 'year':
+        fig_kwargs['x_axis_label'] = 'Year'
+    elif kind == 'weekday':
+        fig_kwargs['x_axis_label'] = 'Day of the week'
+        label_override = [calendar.day_abbr[d] for d in cds.data[kind]]
```

Comment:
We'll need to remember to make these transformations for the csv files when those are implemented. That has me wondering if we should pull some of this big if block into another function or dict. Or we could wait to refactor until we implement csvs. Maybe the latter.

Reply:
If we prefer these labels throughout. I can just move the metrics.calculation module to replace these as the categories after the groupby is complete and right before the metrics are calculated.  That seems to me to be nicer and requires less customization later on. Sound okay or can you think of any reason we might not want to do that?

Reply:
Good idea. I can't think of a reason we wouldn't want to do it. Do we want the mapping to go both ways so that Report's `categories` attribute gets string `'Day of the week'` instead of `'weekday'`? Then, for example, the `weekday` string is internal to metrics.

Reply:
yeah. I think so I like that consistency